### PR TITLE
Set botttom boundary on rellax

### PR DESCRIPTION
**Pull request by jacobraccuia**
By setting the data-rellax-boundary to a valid parent element of the '.rellax', you can stop an element from rellaxing at a certain point.  This is useful if you don't want the element to overlay anymore.

### CODE REVIEW

**Comment by jacobraccuia** [comment]
This could probably be optimized a bit, and or applied to the top as well as the bottom of the boundary wrapper.

**Comment by moeamaya** [comment]
@jacobraccuia This is awesome, a lot of people have been asking me for this! Do you mind adding the top as well?

**Comment by jacobraccuia** [comment]
I added the top!  Please review if you want to change the name of the options (I struggled a bit on what was best) and or to optimize the code better!

File: rellax.js [hunk 1]
```diff
@@ -149,6 +149,10 @@
       var dataSpeed = el.getAttribute( 'data-rellax-speed' );
       var dataZindex = el.getAttribute( 'data-rellax-zindex' ) || 0;
 
+      var bounds = el.getAttribute('data-rellax-bounds') || 'both';
+      var dataWrapper = el.getAttribute('data-rellax-boundary');
+      var boundary = dataWrapper ? el.closest(dataWrapper) : null;
```
Comment (moeamaya) [inline rellax.js]: Should we use closest or just do a general document.querySelector here?

**Comment by jacobraccuia** [comment]
Is closest valid JS? It seems like I mixed in some jQuery habits in my
code.  I think it's important that we use a parent element for the
boundary, but a document.querySelector.closest isn't supported in IE.

The alternative is something like below?

while(thisTag.parentNode.className.indexOf(boundary) === -1)
    {
     thisTag=thisTag.parentNode;
    }



On October 5, 2018 at 10:16:20 AM, Moe Amaya (notifications@github.com)
wrote:

*@moeamaya* commented on this pull request.
------------------------------

In rellax.js
<https://github.com/dixonandmoe/rellax/pull/129#discussion_r223021839>:

> @@ -149,6 +149,10 @@
       var dataSpeed = el.getAttribute( 'data-rellax-speed' );
       var dataZindex = el.getAttribute( 'data-rellax-zindex' ) || 0;

+      var bounds = el.getAttribute('data-rellax-bounds') || 'both';
+      var dataWrapper = el.getAttribute('data-rellax-boundary');
+      var boundary = dataWrapper ? el.closest(dataWrapper) : null;

Should we use closest or just do a general document.querySelector here?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/dixonandmoe/rellax/pull/129#pullrequestreview-162054809>,
or mute the thread

**Comment by alystair** [comment]
Should probably close the PR if a similar one was already integrated

**Pull request closed without merging**